Fix build without ITP

blino opened the pull request:
#### a6bbf5eb1b30d0966a4888c6ad963405854462f4
<pre>
Fix build without ITP
<a href="https://bugs.webkit.org/show_bug.cgi?id=262533">https://bugs.webkit.org/show_bug.cgi?id=262533</a>

Reviewed by Michael Catanzaro.

* Source/WebCore/dom/Element.h:
Drive-by typo fix to rename isSyntheticClick variable as isSyntheticClick

* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::triggerOptionalStorageAccessQuirk const):
Flag isSyntheticClick usage added in 238929@main (98bc1318d160)

* Source/WebCore/platform/network/soup/NetworkStorageSessionSoup.cpp:
(WebCore::NetworkStorageSession::setCookiesFromDOM const):
(WebCore::NetworkStorageSession::getRawCookies const):
(WebCore::cookiesForSession):
Flag shouldRelaxThirdPartyCookieBlocking usage added in 225134@main (664f22663e8d)

* Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.h:
Include HashCountedSet.h to fix build of MessageReceiver objects without ITP

* Source/WebKit/NetworkProcess/ServiceWorker/WebSWServerToContextConnection.cpp:
(WebKit::WebSWServerToContextConnection::fireNotificationEvent):
Flag resourceLoadStatistics() usage added in 255816@main (9404eeeb89ae)

* Source/WebKit/NetworkProcess/cache/NetworkCache.cpp:
Include WebsiteDataType.h for WebsiteDataType::DiskCache usage

* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
Flag RequestStorageAccessResult struct moved in 259893@main (cbe22e46962f)

Canonical link: <a href="https://commits.webkit.org/268975@main">https://commits.webkit.org/268975@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/eaa8ac01936612c32c6a8ee9ab22cfd381c657f4

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/20655 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/21073 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/21728 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/22549 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/19278 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/20892 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/24327 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/21254 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/20612 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/20877 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/20702 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/17957 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/23405 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/17868 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/18763 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/25031 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/18947 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/18940 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/22977 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/19528 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/16590 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/18748 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/5078 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/23085 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/19346 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro commented on Source/WebCore/dom/Element.h:
```diff
@@ -620,7 +620,7 @@ class Element : public ContainerNode {
     void setSavedLayerScrollPosition(const IntPoint&);
 
-    bool dispatchMouseEvent(const PlatformMouseEvent&, const AtomString& eventType, int clickCount = 0, Element* relatedTarget = nullptr, IsSyntheticClick isSyntethicClick = IsSyntheticClick::No);
+    bool dispatchMouseEvent(const PlatformMouseEvent&, const AtomString& eventType, int clickCount = 0, Element* relatedTarget = nullptr, IsSyntheticClick isSyntheticClick = IsSyntheticClick::No);
```
Please placate the style checker:

```
ERROR: Source/WebCore/dom/Element.h:623:  The parameter name "isSyntheticClick" adds no information, so it should be removed.  [readability/parameter_name] [5]
Total errors found: 1 in 7 files
```

blino committed:
Fix build without ITP
https://bugs.webkit.org/show_bug.cgi?id=262533

Reviewed by Michael Catanzaro.

* Source/WebCore/dom/Element.h:
Drive-by typo fix to rename isSyntheticClick variable as isSyntheticClick

* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::triggerOptionalStorageAccessQuirk const):
Flag isSyntheticClick usage added in 238929@main (98bc1318d160)

* Source/WebCore/platform/network/soup/NetworkStorageSessionSoup.cpp:
(WebCore::NetworkStorageSession::setCookiesFromDOM const):
(WebCore::NetworkStorageSession::getRawCookies const):
(WebCore::cookiesForSession):
Flag shouldRelaxThirdPartyCookieBlocking usage added in 225134@main (664f22663e8d)

* Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.h:
Include HashCountedSet.h to fix build of MessageReceiver objects without ITP

* Source/WebKit/NetworkProcess/ServiceWorker/WebSWServerToContextConnection.cpp:
(WebKit::WebSWServerToContextConnection::fireNotificationEvent):
Flag resourceLoadStatistics() usage added in 255816@main (9404eeeb89ae)

* Source/WebKit/NetworkProcess/cache/NetworkCache.cpp:
Include WebsiteDataType.h for WebsiteDataType::DiskCache usage

* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
Flag RequestStorageAccessResult struct moved in 259893@main (cbe22e46962f)

Canonical link: https://commits.webkit.org/268975@main
